AggregateMetrics/AggregateMetrics/AzureWebApp/FlexiblePerformanceCounterGauge.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using Microsoft.ApplicationInsights.DataContracts;
+using System.Net.Http;
+using System.Runtime.Caching;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.ApplicationInsights.Extensibility.AggregateMetrics.AzureWebApp
+{
+    class FlexiblePerformanceCounterGauge
+    {
+        public string GetJson()
+        {
+            HttpClient client = new HttpClient();
+            var t = client.GetStringAsync("http://remoteenvironmentvariables.azurewebsites.net/api/EnvironmentVariables/WEBSITE_COUNTERS_APP/");
+            t.Wait();
+
+            return t.Result;
+        }
+
+        public string GetValue(string name, string json)
+        {
+            string prefix = "\\\""+name+"\\\":";
+            string postfix = ",\\";
+
+            int idx = json.IndexOf(prefix) + prefix.Length;
+            int endIdx = json.IndexOf(postfix, idx);
+
+            return json.Substring(idx, endIdx - idx);
+        }
+
+        public MetricTelemetry GetRawCounterValue(string name)
```

Comment:
I meant make it a separate class that many gauges configured for the different counters will share the one object holding the cache